### PR TITLE
Fcs init speed fix

### PR DESCRIPTION
**Pull request by ulteq**

### CODE REVIEW

File: addons/fcs/functions/fnc_keyUp.sqf [hunk 1]
```diff
@@ -31,9 +31,9 @@ if (_distance == 0) then {
     ] call EFUNC(common,getTargetDistance); // maximum distance: 5000m, 5m precision
 };
 
-private ["_weaponDirection", "_angleTarget"];
-
-_weaponDirection = _vehicle weaponDirection (_vehicle currentWeaponTurret _turret); // @todo doesn't work for sub turrets
+private ["_weapon", "_weaponDirection", "_angleTarget"];
+_weapon = _vehicle currentWeaponTurret _turret;
```
Comment (esteldunedain) [inline addons/fcs/functions/fnc_keyUp.sqf]: The fcs computes the fire solution for all magazines, not only the current weapon so if you switch weapons solution is already computed. We need to find which weapon is used for each magazine.

**Comment by esteldunedain** [comment]
@KoffeinFlummi, @commy2, this seems to be ready now. I'd appreciate you maintainers giving this a look and merging if seems appropiate.

File: addons/fcs/functions/fnc_keyUp.sqf (hunk 1 above)
Comment (commy2) [inline addons/fcs/functions/fnc_keyUp.sqf]: ^ yup
Has to be done for every magazine. Why was this changed here?

Comment (esteldunedain) [inline addons/fcs/functions/fnc_keyUp.sqf]: @commy2, it's already done for every magazine below

File: addons/fcs/functions/fnc_keyUp.sqf [hunk 2]
```diff
@@ -82,6 +83,11 @@ if (time - GVAR(time) > 1 and GVAR(time) != -1 and count _this < 3) then {
     _timeToLive     = getNumber (configFile >> "CfgAmmo" >> _ammoType >> "timeToLive");
     _simulationStep = getNumber (configFile >> "CfgAmmo" >> _ammoType >> "simulationStep");
 
+    _initSpeedCoef = getNumber(configFile >> "CfgWeapons" >> _weapon >> "initSpeed");
+    if (_initSpeedCoef < 0) then {
+        _initSpeed = _initSpeed * -_initSpeedCoef;
+    };
```
Comment (KoffeinFlummi) [inline addons/fcs/functions/fnc_keyUp.sqf]: This doesn't support weapons that have hard coded `initSpeed`s.

```
if (_initSpeedCoef > 0) then {
    _initSpeed = _initSpeedCoef;
};
```

Comment (ulteq) [inline addons/fcs/functions/fnc_keyUp.sqf]: The original code did not support this either:

```_velocityCorrection = (vectorMagnitude velocity _projectile) - getNumber (configFile >> "CfgMagazines" >> _magazine >> "initSpeed");```

It is also questionable whether overriding the magazine `initSpeed` in the weapon `initSpeed` is ever appropriate.

I will change the code accordingly, if we want to support that feature.

Comment (KoffeinFlummi) [inline addons/fcs/functions/fnc_keyUp.sqf]: > The original code did not support this either

I know, that was written before it was a thing ;)

I'd say yes, it's not a good idea on the part of the mod author, but if the game accepts it for ballistic, so should we.

Comment (ulteq) [inline addons/fcs/functions/fnc_keyUp.sqf]: kk

**Comment by ulteq** [comment]
@KoffeinFlummi, @commy2, can we merge this now?

**Comment by jaynus** [comment]
This tests out fine at 1.8km in a comanche. Didn't test anything else. Looks fine though.

**Comment by KoffeinFlummi** [comment]
:+1: from me, up to @commy2
